openraft/src/raft.rs

```diff
@@ -571,6 +571,10 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> Raft<C, N,
         }
         if let Some(handle) = self.inner.raft_handle.lock().await.take() {
             let _ = handle.await?;
+            {
+                let mut guard = self.inner.core_error.lock().unwrap();
+                *guard = Ok(());
+            } // Atomic update core error.
```

Comment:
While `handle.await`, the lock should be held.
Otherwise `get_core_error()` will get an empty result when there is actually an error

Reply:
Maybe consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await

Reply:
Done!